prescient/data/providers/dat_data_provider.py

```diff
@@ -0,0 +1,296 @@
+#  ___________________________________________________________________________
+#
+#  Prescient
+#  Copyright 2020 National Technology & Engineering Solutions of Sandia, LLC
+#  (NTESS). Under the terms of Contract DE-NA0003525 with NTESS, the U.S.
+#  Government retains certain rights in this software.
+#  This software is distributed under the Revised BSD License.
+#  ___________________________________________________________________________
+from __future__ import annotations
+
+from ..data_provider import DataProvider
+from egret.parsers.prescient_dat_parser import get_uc_model, create_model_data_dict_params
+from egret.data.model_data import ModelData as EgretModel
+import os.path
+from datetime import datetime, date, timedelta
+import dateutil.parser
+import copy
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from prescient.simulator.options import Options
+    from typing import Dict, Any
+
+class DatDataProvider():
+    ''' Provides data from pyomo DAT files
+    '''
+
+    def initialize(self, options: Options) -> None:
+        ''' Do one-time initial setup
+        '''
+        self._uc_model_template = get_uc_model()
+        self._instance_directory_name = os.path.join(os.path.expanduser(options.data_directory), 
+                                                     "pyspdir_twostage")
+        self._actuals_by_date = {}
+        self._forecasts_by_date = {}
+        self._first_day = dateutil.parser.parse(options.start_date).date()
+        self._final_day = self._first_day + timedelta(days=options.num_days-1)
+
+    def get_initial_model(self, options:Options, num_time_steps:int) -> EgretModel:
+        ''' Get a model ready to be populated with data
+
+        Returns
+        -------
+        A model object populated with static system information, such as
+        buses and generators, and with time series arrays that are large
+        enough to hold num_time_steps entries.
+
+        Initial values in time time series do not have meaning.
+        '''
+        # Get data for the first simulation day
+        first_day_model = self._get_forecast_by_date(self._first_day)
+
+        # Copy it, making sure we've got the right number of time periods
+        data =_recurse_copy_with_time_series_length(first_day_model.data, num_time_steps)
+        new_model = EgretModel(data)
+        new_model.data['system']['time_keys'] = list(str(i) for i in range(1,num_time_steps+1))
```

Comment:
We be annotating this list with the times (as date-time strings, I think) for which the time series data is for, and this should be a requirement for this interface. That way, when creating the SCED we can do string or datetime comparison operations to know what data to get out of the associated problems. This will be especially necessary for #37.

Reply:
See issue #49